frontend/src/metabase/query_builder/components/notebook/steps/JoinStep.jsx

```diff
@@ -575,6 +593,12 @@ class JoinDimensionPicker extends React.Component {
   open() {
     this._popover.open();
   }
+
+  hasTemporalUnit = fieldRef => {
+    const [, , opts] = fieldRef;
+    return !!opts?.["temporal-unit"];
```

Comment:
let's move `getFieldTemporalUnit` in `frontend/src/metabase-lib/lib/queries/structured/Join.js` to `metabase/lib/join.js` and then we can use it here to do most of this logic

Reply:
Do we want `metabase-lib` to depend on `metabase`? 🤔 I thought we wanted it to be completely independent ideally. At the moment everything is very coupled (and `metabase-lib` even has methods dispatching Redux actions 😛 ), I'm just thinking if we should couple them more